common/sphinx.c

```diff
@@ -379,6 +454,41 @@ static void sphinx_write_frame(u8 *dest, const struct sphinx_hop *hop)
 	memcpy(dest + tal_bytelen(hop->raw_payload), hop->hmac, HMAC_SIZE);
 }
 
+static void sphinx_prefill_stream_xor(u8 *dst, size_t dstlen,
+				      struct secret *shared_secret)
+{
+	u8 padkey[KEY_LEN];
+	u8 prefill[dstlen];
+	generate_key(padkey, "prefill", 7, shared_secret);
+	generate_cipher_stream(prefill, padkey, dstlen);
+	xorbytes(dst, dst, prefill, dstlen);
```

Comment:
Note the new xor_cypher_stream() helper!

Reply:
Had to rebase to get it, but will use that from now on :+1: